httpbp/server.go

```diff
@@ -30,11 +43,12 @@ type httpHandlerFactory struct {
 	middlewares []Middleware
 }
 
-func (f httpHandlerFactory) NewHandler(name string, handle HandlerFunc, middlewares ...Middleware) http.Handler {
-	wrappers := make([]Middleware, 0, len(f.middlewares)+len(middlewares))
+func (f httpHandlerFactory) NewHandler(endpoint Endpoint) http.Handler {
+	wrappers := make([]Middleware, 0, len(f.middlewares)+len(endpoint.Middlewares)+1)
+	wrappers = append(wrappers, SupportedMethods(endpoint.Methods[0], endpoint.Methods[1:]...))
```

Comment:
note, this will panic if you pass in an empty string... let it!  We already check this earlier by calling `endpoint.Validate`